source/core/Array.cpp

```diff
@@ -1121,32 +1121,58 @@ VIREO_FUNCTION_SIGNATURET(ArrayMaxMinInternal, FindArrayMaxMinInstruction)
     AQBlock1* maxValue = minValue;
     if (len == 0)
         maxIndex = minIndex = -1;
+
+    Boolean haveSeenPositiveInfinity = false;
+    Boolean haveSeenNegativeInfinity = false;
     for (IntIndex i = 0; i < len; i++) {
-        Boolean less;
-        snippet->_p0 = arrayIn->BeginAt(i);
+        Boolean shouldUpdateMaxOrMin = false;
+        AQBlock1* currentElement = arrayIn->BeginAt(i);
+        Double currentElementValue = *((Double*)currentElement);
```

Comment:
Treating an element in an array always as double seems to be incorrect. Is the same instruction not getting used for array of other datatypes (Int8, Int16, Int32)? For example, when Int32/Int64 is used with data bits representing the same bits as nan, infinity, -infiinity, we could end up with incorrect results. A series of Int8/Int16 will be interpreted as double.

Reply:
Yeah, this is definitely wrong.  It's not just inf/nan -- the read of memory as double will get the wrong values in ALL cases when the array element type isn't actually doubles.